### PR TITLE
Explicitly build against libevent 2.1.10 and libevent 2.1.12

### PR DESCRIPTION
**Pull request by jjerphan**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Reasons for this change:

folly currently is built only against `libevent==2.1.12`. This was the case in the past, but it stopped with https://github.com/conda-forge/folly-feedstock/commit/5a065591e7b7530ecc56baff71f88284640f8a08.

Some packages (namely ArcticDB) need its dependencies (which include folly) to be built against `libevent==2.1.10`. 

This explicitly built folly against `libevent==2.1.10` and `libevent==2.1.12`.

This zips keys not to build against all the Cartesian product of dependencies.

### CODE REVIEW

**Comment by jjerphan** [comment]
@conda-forge-admin, please rerender

**Comment by jjerphan** [comment]
@conda-forge-admin, please rerender

**Comment by jjerphan** [comment]
@conda-forge-admin, please rerender

**Comment by jjerphan** [comment]
Do you think this is acceptable, @xhochy? :pray:

**Comment by jjerphan** [comment]
Some customers of ArcticDB only have constrained environments in which libevent 2.1.10 has to be used unfortunately.

**Comment by jjerphan** [comment]
And as for fmt, fmt 10 changed their API IIRC, and ArcticDB still relies on fmt 9. Using fmt 10 is a question of prioritization, and it is not a priority yet, I am afraid.

**Comment by xhochy** [comment]
> Some customers of ArcticDB only have constrained environments in which libevent 2.1.10 has to be used unfortunately.

Is there any good reason for this? If so, we may add it to the global pinning which could ease the process for you overall.

**Comment by jjerphan** [comment]
> Is there any good reason for this?

I am trying to get this information and to see if they could update their internal setup to us latest version of packages. What I am sure of is that they only (can) use python 3.7 and python 3.9 environments which generally come with a significant number of Python packages for data analysis, including the Apache Arrow and Jupyter stacks. This might constraints the usage of previous versions of packages, and of their dependencies (including libevent).

> If so, we may add it to the global pinning which could ease the process for you overall.

Oh, I appreciate the proposal, yet I think this would put extra unnecessary constraints on conda-forge as a whole. Defining various at the level of folly and ArcticDB is sufficient, and is not much of a cost for us.

**Comment by jjerphan** [comment]
If needed, we can restrict those new build variants to `[unix]`. What do you think?

**Comment by jjerphan** [comment]
@xhochy: what do you think? Having those variants is the crux for us to solve ABI instability problems. :pray:

**Comment by xhochy** [comment]
I'm fine with the variants. My messages were targeted at getting the long-term issue solved (which the global pinning and fast migrations in conda-forge should theoretically fix).

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -6,5 +6,13 @@ folly_build_ext:
...
   - "jemalloc"  # [not win]
 fmt:
-  - "10"
   - "9"
+  - "9"
+  - "10"
+libevent:
+  - "2.1.10"
+  - "2.1.12"
+  - "2.1.12"
```
Comment (jjerphan) [inline recipe/conda_build_config.yaml:15]: According to [the specification of `zip_keys`](https://docs.conda.io/projects/conda-build/en/stable/resources/variants.html#coupling-keys), this must build for:
 - fmt 9 + libevent 2.1.10
 - fmt 9 + libevent 2.1.12
 - fmt 10 + libevent 2.1.12

but [the recipe has not and instead have built for](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=768985&view=results):
 - fmt 10 + libevent 2.1.10
 - fmt 10 + libevent 2.1.12
 - fmt 9 + libevent 2.1.12

Comment (jjerphan) [reply recipe/conda_build_config.yaml:15]: A work-around is proposed by https://github.com/conda-forge/folly-feedstock/pull/137/.

Comment (jjerphan) [reply recipe/conda_build_config.yaml:15]: The issue is tracked by https://github.com/conda/conda-build/issues/4987.

**Pull request merged**